py/torch_tensorrt/dynamo/conversion/_conversion.py

```diff
@@ -63,6 +68,128 @@ def interpret_module_to_result(
         SerializedInterpreterResult
     """
 
+    def _insert_engine_to_cache(
+        hash_val: str, interpreter_result: TRTInterpreterResult
+    ) -> None:  # type: ignore[unused-ignore]
+        # Cache the weight-stripped engine regardless of the `strip_engine_weights` setting
+        if engine_cache.check(hash_val) is not None:  # type: ignore[union-attr]
+            logger.info(f"Engine already exists in cache for hash: {hash_val}")
+            return
+        if not settings.strip_engine_weights:
+            # set EXCLUDE_WEIGHTS flag to strip weights
+            serialization_config = (
+                interpreter_result.engine.create_serialization_config()
+            )
+            serialization_config.set_flag(trt.SerializationFlag.EXCLUDE_WEIGHTS)
+            weight_stripped_serialized_engine = (
+                interpreter_result.engine.serialize_with_config(serialization_config)
+            )
+        else:
+            weight_stripped_serialized_engine = interpreter_result.engine.serialize()
+
+        # Insert weight-stripped engine to cache
+        engine_cache.insert(  # type: ignore[union-attr]
+            hash_val,
+            (
+                weight_stripped_serialized_engine,
+                interpreter_result.input_names,
+                interpreter_result.output_names,
+                inputs,
+                settings,
+                interpreter_result.weight_name_map,
+                interpreter_result.requires_output_allocator,
+            ),
+        )
+        logger.info(f"Engine was successfully inserted into cache for hash: {hash_val}")
+
+    @needs_refit  # type: ignore[misc]
```

Comment:
Should the insert and extract both be needs refit? 

Also shouldnt this gracefully pass through vs the typically unimplemented error?